libsplinter/src/network/auth/authorization/challenge/handlers.rs

```diff
@@ -323,7 +323,7 @@ impl Handler for AuthChallengeSubmitRequestHandler {
                 .verify(
                     &self.nonce,
                     &Signature::new(request.signature.to_vec()),
-                    &PublicKey::new(request.public_key.to_vec()),
+                    &PublicKey::new(request.public_key.clone().into_bytes()),
```

Comment:
What’s being cloned here?

Reply:
Its the splinter::public_key::PublicKey stored in the request. into_bytes consumes the public key so it needs to be cloned to convert it to cylinder::PublicKey. Could be changed to `&PublicKey::new(request.public_key.as_slice().to_vec()),`  instead